Add polyfill for getStats in adapter.js.

milkypostman opened the pull request:
The Chrome version of getStats produces a cryptic data structure. This
wraps the getStats call so that it comes out in the W3C standard
structure.

juberti commented on src/js/adapter_test.js:
```diff
@@ -0,0 +1,82 @@
+var AdapterTest = new TestCase('AdapterTest');
+
+AdapterTest.prototype.setUp = function() {
+  window.console.log("Setting stuff up!");
```
I assume you meant to remove this?

juberti commented on src/js/test_preload.js:
```diff
@@ -0,0 +1,21 @@
+ *  tree.
+ */
+
+/* This file is used to simply extract the Chrome bits somewhere we can test. */
```
Please dumb this down to I can understand what is happening here

juberti replied:
Please dumb this down so I can understand what is happening here

juberti commented on src/js/adapter.js:
```diff
@@ -181,7 +181,33 @@ if (navigator.mozGetUserMedia) {
-    return new webkitRTCPeerConnection(pcConfig, pcConstraints);
+    var pc = new webkitRTCPeerConnection(pcConfig, pcConstraints);
+    var origGetStats = pc.getStats.bind(pc);
+    pc.getStats = function(selector, successCallback) {
```
To avoid breaking all stats users, should we just leave the stats untouched if the old argument order was used?

milkypostman replied:
Hrm. That's a thought. It's going to ugly up the code a bit because we'll have to do type checking on the arguments to see if one is a function or not. Another option would be to keep the old function under a different name.

milkypostman commented:
@juberti have addressed comments and reworked the testing setup. This is now specialized to Chrome but allows us to add Firefox tests in the future. Each needs it own *setup* because we are essentially faking out the differing parts.

KaptenJansson commented:
milkypostman@ when do you think you can move this to the adapter repo?

milkypostman commented:
i just need to carve out time to do it again. i'll try to get this going.

On Mon, Jun 1, 2015 at 11:39 PM Christoffer Jansson <
notifications@github.com> wrote:

> milkypostman@ when do you think you can move this to the adapter repo?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/webrtc/samples/pull/484#issuecomment-107826656>.
>

milkypostman commented:
opened new pr in webrtc/adapter#46
